packages/experimental/src/global/js/utils/unwrap-if-fragment.js

```diff
@@ -1,14 +1,31 @@
 import React from 'react';
 
+/**
+ * unwrapIfFragment - walks the passed children unwrapping fragments
+ * to create a flat array of all children
+ */
 const unwrapIfFragment = (children) => {
   const newChildArray = [];
-  const realChildren =
-    children && children.type === React.Fragment
-      ? children.props.children
-      : children;
-  for (let index in realChildren) {
-    newChildArray.push(realChildren[index]);
+
+  const isFragment = (item) => item && item.type === React.Fragment;
+
+  const addChildren = (_children) => {
+    for (let index in _children) {
+      if (isFragment(_children[index])) {
+        addChildren(_children[index].props.children);
+      } else {
+        newChildArray.push(_children[index]);
+      }
+    }
+  };
+
+  // nothing, one fragment, one or more children
+  if (isFragment(children)) {
```

Comment:
This could be abbreviated to `addChildren([children]);`. Just a thought.

Reply:
> This could be abbreviated to `addChildren([children]);`. Just a thought.

It could but would perhaps not be clear why on this line that addChildren() will unpack the array.